src/sdk/service/iam-service.ts

```diff
@@ -0,0 +1,178 @@
+    }
+    const policyOptional = role.getPolicy(entity.uuid);
+    if (policyOptional !== null) {
+      return policyOptional;
+    } else if (entity.parentUuid !== null) {
+      return this.findFirstRelevantPolicy(companyInventory,
+        companyInventory.getEntityByUuid(entity.parentUuid), role);
+    } else {
+      return null;
+    }
+  }
+
+  /**
+   * Derive effective policy from an ancestor.
+   * @param {CompanyInventory} companyInventory
+   * @param {Policy} ancestorPolicy
+   * @param {InventoryEntity | undefined} target
+   * @returns {Policy | null}
+   */
+  private deriveEffectivePolicyFromAncestor(companyInventory: CompanyInventory, ancestorPolicy: Policy,
+                                            target: InventoryEntity | undefined): Policy | null {
+    if (target === undefined) {
+      return null;
+    } else if (ancestorPolicy.entityDomain === target.type) {
+      return ancestorPolicy;
+    }
+    const targetParent = companyInventory.getEntityByUuid(target.parentUuid);
+    const directParentPolicy: Policy | null =
+      this.deriveEffectivePolicyFromAncestor(companyInventory, ancestorPolicy, targetParent);
+    if (directParentPolicy === null) {
+      return null;
+    }
+    switch (directParentPolicy.type) {
+      case 'ADMIN':
+        return new PolicyBuilder(target.uuid, target.type, 'ADMIN').build();
+      case 'READ_ONLY':
+        return new PolicyBuilder(target.uuid, target.type, 'READ_ONLY').build();
+      case 'CUSTOM':
+      default:
+        const builder: PolicyBuilder = new PolicyBuilder(target.uuid, target.type, 'CUSTOM');
+        const permissions: Array<Permission> | null =
+          this._permissionService.getImpliedPermissions(directParentPolicy.permissions);
+        if (permissions !== null) {
+          for (const p of permissions) {
+            if (p.domain === target.type) {
+              builder.addPermission(p.permissionType);
+            }
+          }
+        }
+        return builder.build();
+    }
+  }
+
+  /**
+   * Check whether or not an entity is accessible publicly.
+   * @param {string} entityUuid
+   * @param {PermissionType} permission
+   * @returns {boolean}
+   */
+  private isPubliclyAccessible(entityUuid: string, permission: PermissionType): boolean {
+    const permissionObject: Permission | undefined = this._permissionService.permissions.get(permission);
+    if (permissionObject) {
+      // handle public catalogs, media, and templates
+      switch (permissionObject.domain) {
+        case 'ILAND_CLOUD_CATALOG':
+          return (entityUuid.includes('public:') && permission === 'VIEW_ILAND_CLOUD_CATALOG');
+        case 'ILAND_CLOUD_MEDIA':
+          return (entityUuid.includes('public:') && (permission === 'VIEW_ILAND_CLOUD_MEDIA' ||
+            permission === 'CLONE_DOWNLOAD_ILAND_CLOUD_MEDIA'));
+        case 'ILAND_CLOUD_VAPP_TEMPLATE':
+          return (entityUuid.includes('public:') && (permission === 'VIEW_ILAND_CLOUD_VAPP_TEMPLATE' ||
```

Comment:
You shouldn't be checking individual permissions on these. Instead you should be checking whether the specified permission's AccessType is READ

Reply:
Ok. I'll update this.